.github/workflows/nodejs_bindings.yml

```diff
@@ -0,0 +1,319 @@
+            mkdir -p ../valhalla-npm-package/${os}/${arch}
+            mv valhalla-nodejs-${platform_name}-${arch}/valhalla_node.node ../valhalla-npm-package/${os}/${arch}/
+            mv valhalla-nodejs-${platform_name}-${arch}/valhalla_* ../valhalla-npm-package/${os}/${arch}/ 2>/dev/null || true
+            mv valhalla-nodejs-${platform_name}-${arch}/lib ../valhalla-npm-package/${os}/${arch}/
+          done
+          
+          popd # artifacts
+          
+          echo "[INFO] Package structure:"
+          tree -L 5 valhalla-npm-package/ 
+          
+          echo "[INFO] Creating NPM package..."
+          pushd valhalla-npm-package
+          npm pack
+          popd # valhalla-npm-package
+
+      - name: Upload NPM package
+        uses: actions/upload-artifact@v4
+        with:
+          name: valhalla-npm-package
+          path: valhalla-npm-package/*.tgz
+
+  verify_nodejs_package:
+    name: Verify NPM package (${{ matrix.os }}-${{ matrix.arch }})
+    runs-on: ${{ matrix.runner }}
+    needs: [package_nodejs_bindings]
+    container: ${{ matrix.container }}
+    strategy:
+      matrix:
+        include:
+          - os: linux
+            arch: x64
+            runner: ubuntu-latest
+            container: node:18
+          - os: linux
+            arch: arm64
+            runner: ubuntu-24.04-arm
+            container: node:18
+          - os: macos
+            arch: arm64
+            runner: macos-14
+            container: null
+    steps:
+      - name: Install Node.js (macOS)
+        if: matrix.os == 'macos'
+        uses: actions/setup-node@v4
+        with:
+          node-version: '18'
+
+      - name: Checkout test files
+        uses: actions/checkout@v4
+        with:
+          sparse-checkout: |
+            src/bindings/nodejs/test/
+            test/data/utrecht_netherlands.osm.pbf
+          sparse-checkout-cone-mode: false
+
+      - name: Download NPM package
+        uses: actions/download-artifact@v4
+        with:
+          name: valhalla-npm-package
+      
+      - name: Install package
+        run: npm install *.tgz
+
+      - name: Test package import (CommonJS)
+        run: node --test src/bindings/nodejs/test/test-commonjs.js
+      
+      - name: Test package import (ESM)
+        run: node --test src/bindings/nodejs/test/test-esm.mjs
+      
+      - name: Test Valhalla CLI
+        run: |
+          npx valhalla --version
+          npx valhalla --help
+          npx valhalla build_tiles -h
+          npx valhalla valhalla_build_tiles -h
+
+      - name: Integration test
+        run: |
+          echo "[INFO] Creating Valhalla config..."
+          mkdir -p valhalla_tiles
+          npx valhalla build_config --mjolnir-tile-dir valhalla_tiles --mjolnir-tile-extract valhalla_tiles.tar > valhalla.json
+
+          echo "[INFO] Building tiles from Utrecht OSM data..."
+          npx valhalla build_tiles -c valhalla.json test/data/utrecht_netherlands.osm.pbf >/dev/null 2>&1
```

Comment:
argh, of course we can do that for the python bindings too. good point, much better than what we have there currently